pkg/probe/grpc/grpc.go

```diff
@@ -57,6 +57,9 @@ func (p grpcProber) Probe(host, service string, port int, timeout time.Duration)
 		grpc.WithUserAgent(fmt.Sprintf("kube-probe/%s.%s", v.Major, v.Minor)),
 		grpc.WithBlock(),
 		grpc.WithTransportCredentials(insecure.NewCredentials()), //credentials are currently not supported
+		grpc.WithContextDialer(func(ctx context.Context, addr string) (net.Conn, error) {
```

Comment:
not a lot of experience with grpc , but this looks good to me 

https://pkg.go.dev/google.golang.org/grpc#WithContextDialer